### PR TITLE
cleanup: Add warning in amountMath.coerce()

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
Add a warning for a (presumed) common error in amountMath.coerce()

### CODE REVIEW

**Comment by Chris-Hibbert** [comment]
Anyone have a theory about what Conventional Commits would want me to say for this change?

It's not a customer-visible change in behavior; it doesn't change any APIs; it's not in response to a bug report; I could call it 'doc' because it's for developer support; but it's not part of the doc suite, nor visible as documentation anywhere.

File: packages/ERTP/src/amountMath.js [hunk 1]
```diff
@@ -83,6 +83,10 @@ function makeAmountMath(brand, mathHelpersName) {
         return allegedAmount;
       }
       const { brand: allegedBrand, extent } = allegedAmount;
+      assert(
+        allegedBrand !== null,
```
Comment (katelynsills) [inline packages/ERTP/src/amountMath.js]: This should be `undefined` and not null. Can we add a test to hit this case to ensure it's catching the right thing?

**Review by katelynsills — CHANGES_REQUESTED** [review]

File: packages/ERTP/src/amountMath.js [hunk 2]
```diff
@@ -83,6 +83,10 @@ function makeAmountMath(brand, mathHelpersName) {
         return allegedAmount;
       }
       const { brand: allegedBrand, extent } = allegedAmount;
+      assert(
+        allegedBrand !== undefined,
+        details`alleged brand is null. Did you pass an extent rather than an amount?`,
```
Comment (katelynsills) [inline packages/ERTP/src/amountMath.js]: Can we change the error message to say undefined rather than null?

Comment (Chris-Hibbert) [reply packages/ERTP/src/amountMath.js]: Yup. Good catch.

Comment (Chris-Hibbert) [reply packages/ERTP/src/amountMath.js]: Yes, of course.

**Comment by katelynsills** [comment]
> Anyone have a theory about what Conventional Commits would want me to say for this change?
> 
> It's not a customer-visible change in behavior; it doesn't change any APIs; it's not in response to a bug report; I could call it 'doc' because it's for developer support; but it's not part of the doc suite, nor visible as documentation anywhere.

Maybe 'chore' or 'refactor'? Fix sounds ok to me too

**Review by katelynsills — APPROVED** [review]
LGTM! Thanks, this is a fantastic usability improvement

**Pull request merged**